client/components/close-on-escape/index.jsx

```diff
@@ -0,0 +1,71 @@
+/**
+ * External dependencies
+ */
+import { filter, includes, isEmpty, last, noop } from 'lodash';
+import PropTypes from 'prop-types';
+import { Component } from 'react';
+
+const ESC_KEY_CODE = 27;
+let components = [];
+
+function onKeydown( event ) {
+	if (
+		! isEmpty( components ) &&
+		event.keyCode === ESC_KEY_CODE &&
```

Comment:
Although `event.keyCode` is widely supported and may stay with us forever, [it's actually deprecated now](https://developer.mozilla.org/en-US/docs/Web/API/KeyboardEvent/keyCode). React [does a great job of normalizing `event.key`](https://github.com/facebook/react/blob/master/src/renderers/dom/shared/utils/getEventKey.js) and I believe it's the preferred way to test for "special" keys. I think we should use that here and maybe later update our whole codebase for similar cases.

Reply:
Oh sad, just noticed this is outside of React event system. In that case, we are probably better with keyCode. Ignore me

Reply:
Good call out though - we could always put some logic around this if the need arises :)